pkg/bosh/converter/container_factory.go

```diff
@@ -24,7 +24,13 @@ const (
 )
 
 // ContainerFactory builds Kubernetes containers from BOSH jobs.
-type ContainerFactory struct {
+type ContainerFactory interface {
```

Comment:
Can we follow the advice from https://github.com/golang/go/wiki/CodeReviewComments#interfaces and declare the interface on the consumer side? The decoupling is nice and we see directly which methods from the interface are used by every consumer.

If you decide to keep the interface here, we need a check like `var _ reconcile.Reconciler = &ReconcileBOSHDeployment{}`. These seem to be missing for several of our interfaces :?